mungegithub/pulls/blunderbuss.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pulls
+
+import (
+	"flag"
+	"math/rand"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/yaml"
+)
+
+// A BlunderbussConfig maps a set of file prefixes to a set of owner names (github users)
+type BlunderbussConfig struct {
+	PrefixMap map[string][]string `json:"prefixMap,omitempty" yaml:"prefixMap,omitempty"`
+}
+
+func (b *BlunderbussConfig) FindOwners(filename string) []string {
+	owners := util.StringSet{}
+	for prefix, ownersList := range b.PrefixMap {
+		if strings.HasPrefix(filename, prefix) {
+			owners.Insert(ownersList...)
+		}
+	}
+	return owners.List()
+}
+
+type BlunderbussMunger struct {
+	config *BlunderbussConfig
+}
+
+var blunderbussConfig = flag.String("blunderbuss-config", "", "Path to the blunderbuss config file")
```

Comment:
as a future addition, the blunderbuss-config could be loaded from the repository lake the .travis.yml